src/OrchardCore.Modules/Orchard.Users/Services/UserStore.cs

```diff
@@ -322,19 +346,41 @@ public Task<IList<string>> GetRolesAsync(User user, CancellationToken cancellati
             return Task.FromResult<IList<string>>(user.RoleNames);
         }
 
-        public Task<bool> IsInRoleAsync(User user, string roleName, CancellationToken cancellationToken)
+        public async Task<bool> IsInRoleAsync(User user, string normalizedRoleName, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
No. I discussed with the Kestrel dev when I found some logs during my stress test related to cancellation token. The issue is that some connections get cancelled, and these services are using the web connection cancellation token. It fills up the log for nothing. And in this case, it's simpler to finish the request than detect it an stop processing. It's not like these services are heavy on the cpu.